src/index/CompressedRelation.cpp

```diff
@@ -1085,25 +1090,27 @@ CompressedRelationReader::getRelevantBlocks(
     return blockA.lastTriple_ < blockB.firstTriple_;
   };
 
-  return ql::ranges::equal_range(blockMetadata, key, comp);
-}
-
-// _____________________________________________________________________________
-std::span<const CompressedBlockMetadata>
-CompressedRelationReader::getBlocksFromMetadata(
-    const ScanSpecAndBlocks& metadata) {
-  return getRelevantBlocks(metadata.scanSpec_, metadata.blockMetadata_);
+  BlockMetadataRanges resultBlocks;
+  ql::ranges::for_each(
+      blockMetadata, [&resultBlocks, &key,
+                      &comp](const BlockMetadataRange& blockMetadataSubrange) {
+        auto result = ql::ranges::equal_range(blockMetadataSubrange, key, comp);
+        if (result.begin() != result.end()) {
+          resultBlocks.emplace_back(result.begin(), result.end());
+        }
+      });
+  return resultBlocks;
 }
 
 // _____________________________________________________________________________
 auto CompressedRelationReader::getFirstAndLastTriple(
     const CompressedRelationReader::ScanSpecAndBlocks& metadataAndBlocks,
     [[maybe_unused]] const LocatedTriplesPerBlock& locatedTriplesPerBlock) const
     -> std::optional<ScanSpecAndBlocksAndBounds::FirstAndLastTriple> {
-  auto relevantBlocks = getBlocksFromMetadata(metadataAndBlocks);
-  if (relevantBlocks.empty()) {
+  if (metadataAndBlocks.sizeBlockMetadata_ == 0) {
     return std::nullopt;
   }
+  auto blocks = metadataAndBlocks.getBlockMetadataView();
```

Comment:
prefer `const auto&` to make the code agnostig on "is that a cheap or an expensive type".